### DISCUSSION TITLE
(Suggestion) Implementing claim kick commands

### DISCUSSION THREAD

**Discussion by MithrandirCraft** [GriefPrevention/GriefPrevention#224] [Ideas]
The suggestion - Implementing commands such as:

/playerkick <player/all> 
Meant for kicking players from a territory. Banning could also be considered, however this could be highly exploitable. The command could either send the player in question to the world spawn, or to the closest external block location to the foreighn claim he was in. Would also have a preconfigurable usage cooldown.

/entitykick
Meant for removing inactive and invincible foreign pets which are just standing there on someone's own territory, and cannot be removed in any method imaginable.

----------------------

Reasoning behind this suggestion: My Survival server is small, but has been on for more than two years.
The most common complaints are of random people acting stupid on other people's claims. And inactive entities on foreighn claims.
Claim owners have a right to decide who and what can stay on their territories.

I imagine this could conflict with the siege mechanics (Which I don't use), but could be an inactive feature during sieges.

Spigot: 1.12.1
GP: 16.7.1

**Comment by RoboMWM** [comment]
Pretty sure I've seen a plugin that has already implemented this, though I have not tested it. If you want me to get a link to it just let me know.

**Comment by MithrandirCraft** [comment]
Yes please! :) I don't mind if it's an addon.

**Comment by MithrandirCraft** [comment]
https://www.spigotmc.org/resources/kickfromclaim.17371/

May be it? Not updated but could still work.
However, It only kicks players. What to do with abandoned foreign pets?

**Comment by RoboMWM** [comment]
Not sure if any exists, could probably suggest that the author adds that as well.

**Comment by MithrandirCraft** [comment]
I will. However this doesn't seem outside of your plugin's scope.
You have terrain regeneration, why not more player impermeability features such as this?

As far as I know, there are also still ways of killing players using lava, suffocation and other methods, and there are always complaints about inactive foreign claims near other claims, or people placing scrap near other people's territories just as a bother. I'll provably be ticketing on those issues soon.

GP creates a pretty violence free enviroment, but it's still not perfect.

**Comment by RoboMWM** [comment]
> You have terrain regeneration, why not more player impermeability features such as this?

Although this will not be in version 20, this was largely added to deal with "nature grief." Not any particular claim (it in facts ensures to ignore claims). This tool was likely added since there's no easy way to "prevent" natural/terrain grief, it also should also be easy to fix, since unlike claims, nobody really cares about its contents (it's more of an "appearance/aesthetics" grief I suppose).

The only "annoying" thing I can think of is players preventing the claim owner from placing a block in a particular location, which is arguably more of an issue in creative servers. However, the idea of "flags" in a claim has always been avoided in this plugin for simplicity (via the three trust levels). Flags and other additional functionality can and is offered in addons. (One such plugin includes "banning" access to a claim, a-la perma claim kick.)

>  killing players using lava

There's an option for this, regarding placement of lava that is.

As for the rest, I'm not really sure what you're trying to get at. Inactive claim expiration can be configured, and you can decide to restrict building to only claims (or encourage your players to claim a tad more than what they intend to use).

**Comment by TheSenseiNL** [comment]
i use /kfc on my server and it works like a charmto kick people to the outside of your claim. tested it with 1.8 till 1.12
